src/java/io/bazel/rulesscala/scalac/ReportableMainClass.java

```diff
@@ -4,41 +4,44 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import scala.tools.nsc.reporters.Reporter;
 import scala.tools.nsc.Settings;
 import scala.tools.nsc.reporters.ConsoleReporter;
 import scala.tools.nsc.MainClass;
 import scala.tools.nsc.Global;
 
-
-public class ReportableMainClass extends MainClass{
+public class ReportableMainClass extends MainClass {
     private Global compiler;
     private final CompileOptions ops;
 
-    public ReportableMainClass(CompileOptions ops){
+    public ReportableMainClass(CompileOptions ops) {
         this.ops = ops;
     }
 
     @Override
     public Global newCompiler() {
+        if (!ops.enableDiagnosticsReport) {
+            createDiagnosticsFile();
```

Comment:
This looks a bit confusing to me. Diagnostics file is created even when diagnostics is not enabled.
Diagnostics file is created in two branches but doesn't seem to be used. Maybe it's a wrong place to create this file?

Reply:
Diagnostics is still defined as rule output, so it expects this file to be produced - this is how original implementation works.